charts/kube-prometheus-stack/templates/prometheus/networkpolicy.yaml

```diff
@@ -0,0 +1,32 @@
+{{- if .Values.prometheus.networkPolicy.enabled }}
+apiVersion: {{ template "kube-prometheus-stack.prometheus.networkPolicy.apiVersion" . }}
+kind: NetworkPolicy
+metadata:
+  labels:
+    app: {{ template "kube-prometheus-stack.name" . }}-prometheus
+{{- include "kube-prometheus-stack.labels" . | indent 4 }}   
+  name: {{ template "kube-prometheus-stack.fullname" . }}-prometheus
+  namespace: {{ template "kube-prometheus-stack.namespace" . }}
+spec:
+  egress:
+    {{- if .Values.prometheus.networkPolicy.egress }}
+    {{- toYaml .Values.prometheus.networkPolicy.egress | nindent 4 }}
+    {{- else }}
```

Comment:
default can be defined in the values yaml.
i would likely not use "allow all" by default but give a set of reasonable policies.

Reply:
Pushed a new version with limited default ingress

Reply:
We might want to make the whole .spec part configuralbe by using "{{ toYaml ...}}"?
Just in case somebody likes to change selectors too?

egress default for the operator would be kube api. 
prometheus should be able to reach 9090 for the scrapes. 
Not sure if i miss something?

We could also use deny all as default so everybody has to care individually if networks policy is enabled.

Reply:
I would think that someone wanting to enable the netpol would have to change the egress anyway unless we set it to allow all. As enablement is a conscious choice, let's limits it to 9090, the kube api and alertmanager then. 
I'll try to figure out how :D 

Regarding the selectors, maybe we allow them to override them for now but not the whole spec? Otherwise we would need to do the same change for prometheus-operator network policies :)

Reply:
Just wanted to go the easy path instead defining overrides for every available option.
In the end i don't much of an opinion here.
Maybe wait for some feedback of the charts maintainers.

Reply:
@scottrigby @Xtigyro does any one of you have ideas?

Otherwise let's make it deny  or allow all and users will override it to fit their needs

Reply:
@monotek I made the egress deny all by default, letting the users override it for now. This could be revisited later